src/components/Counter/Counter.jsx

```diff
@@ -7,51 +7,80 @@ const colors = {
   red: 'rgb(239, 68, 68)',
 };
 
+import { useReducer } from 'react';
+
+const initialState = { count: 0, color: colors.yellow };
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'INCREMENT_COUNT':
+      return {
+        count: state.count + 1,
+        color: state.color,
+      };
+    case 'DECREMENT_COUNT':
+      return {
+        count: state.count - 1,
+        color: state.color,
+      };
+    case 'RESET_COUNT':
+      return {
+        count: (state.count = 0),
+        color: state.color,
+      };
+    case 'CHANGE_COLOR':
```

Comment:
If you wanted, you could take this section out and pass the color directly to the color prop, similar to like line 12.